src/pages/Home.tsx

```diff
@@ -27,8 +27,18 @@ export default function Home() {
       throw error;
     }
   }, []);
+  const [cancelWaitForChecks, setCancelWaitForChecks] = useState<() => void>(() => {
+    return () => {};
+  });
+  useEffect(() => {
+    return () => {
+      cancelWaitForChecks();
+    };
+  }, [cancelWaitForChecks]);
   const [completedChecksState, getCompletedChecks] = useAsyncFn(async (names?: string[]) => {
-    await api.waitForChecks(names);
+    const { promise, cancel } = await api.waitForChecks(names);
+    setCancelWaitForChecks(() => cancel);
```

Comment:
Would it work to just pass the `cancel` function instead of wrapping it?
```suggestion
    setCancelWaitForChecks(cancel);
```